modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -70,6 +70,7 @@ public Settings getAdditionalIndexSettings(
             // First backing index is created and then data stream is rolled over (in a single cluster state update).
             // So at this point we can't check index_mode==time_series,
             // so checking that index_mode==null|standard and templateIndexMode == TIME_SERIES
+            boolean isTimeSeries = templateIndexMode == IndexMode.TIME_SERIES;
```

Comment:
Check for null?

Reply:
if `templateIndexMode` is null then this evaluates false, which is expected. So I think this is good?